include/mbedtls/asn1.h

```diff
@@ -240,7 +240,9 @@ int mbedtls_asn1_get_len(unsigned char **p,
  * \param p     On entry, \c *p points to the start of the ASN.1 element.
  *              On successful completion, \c *p points to the first byte
  *              after the length, i.e. the first byte of the content.
- *              On error, the value of \c *p is undefined.
+ *              If #MBEDTLS_ERR_ASN1_UNEXPECTED_TAG is returned, \c *p
```

Comment:
Please add unit tests.

Reply:
If this property did not hold then at least the X.509 code would be broken.

Reply:
The ASN.1 module was in a separate project from X.509 at some point in the past, and might end up in a separate project at some point in the future. So we don't want to rely on X.509 tests for the correctness of the ASN.1 code. (That, and debugging core ASN.1 functions via complex X.509 processing is not fun.)

Reply:
That makes sense.  The hardest part of making the ASN.1 parser stricter was not actually changing the code, but rather fixing all of the tests that relied on the old behavior.